azure-iot-hub/tests/test_iothub_registry_manager.py

```diff
@@ -0,0 +1,881 @@
+            == mock_module_constructor.return_value
+        )
+
+
+@pytest.mark.describe("IoTHubRegistryManager - .update_module_with_certificate_authority()")
+class TestUpdateModuleWithCA(object):
+    @pytest.mark.it("initializes module with device id, module id, managed_by, status and ca auth")
+    def test_initializes_device_with_kwargs_for_certificate_authority(
+        self, mock_module_constructor, mock_service_operations, iothub_registry_manager
+    ):
+        iothub_registry_manager.update_module_with_certificate_authority(
+            device_id=fake_device_id,
+            module_id=fake_module_id,
+            etag=fake_etag,
+            status=fake_status,
+            managed_by=fake_managed_by,
+        )
+
+        assert mock_module_constructor.call_count == 1
+        assert mock_module_constructor.call_args[1]["module_id"] == fake_module_id
+        assert mock_module_constructor.call_args[1]["managed_by"] == fake_managed_by
+        assert mock_module_constructor.call_args[1]["device_id"] == fake_device_id
+        assert mock_module_constructor.call_args[1]["status"] == fake_status
+        assert isinstance(
+            mock_module_constructor.call_args[1]["authentication"], AuthenticationMechanism
+        )
+        auth_mechanism = mock_module_constructor.call_args[1]["authentication"]
+        assert auth_mechanism.type == "certificateAuthority"
+        assert auth_mechanism.x509_thumbprint is None
+        assert auth_mechanism.symmetric_key is None
+
+    @pytest.mark.it(
+        "calls method from service operations with device id, module id and previously constructed module"
+    )
+    def test_calls_create_or_update_device_for_certificate_authority(
+        self, mock_module_constructor, mock_service_operations, iothub_registry_manager
+    ):
+        iothub_registry_manager.update_module_with_certificate_authority(
+            device_id=fake_device_id,
+            module_id=fake_module_id,
+            etag=fake_etag,
+            status=fake_status,
+            managed_by=fake_managed_by,
+        )
+
+        assert mock_service_operations.create_or_update_module.call_count == 1
+        assert mock_service_operations.create_or_update_module.call_args[0][0] == fake_device_id
+        assert mock_service_operations.create_or_update_module.call_args[0][1] == fake_module_id
+        assert (
+            mock_service_operations.create_or_update_module.call_args[0][2]
+            == mock_module_constructor.return_value
+        )
+
+
+@pytest.mark.describe("IoTHubRegistryManager -- .get_module()")
+class TestGetModule(object):
+    @pytest.mark.it("gets module from service for provided device id and module id")
+    def test_get_module(self, mocker, mock_service_operations, iothub_registry_manager):
+        iothub_registry_manager.get_module(fake_device_id, fake_module_id)
+
+        assert mock_service_operations.get_module.call_args == mocker.call(
```

Comment:
Make sure to add an assertion on the call count as well. Here and everywhere that it isn't done.